lib/screens/v2/candidate_generator/bus_eink.ex

```diff
@@ -32,19 +32,23 @@ defmodule Screens.V2.CandidateGenerator.BusEink do
   end
 
   @impl CandidateGenerator
+  # credo:disable-for-next-line Credo.Check.Design.DuplicatedCode
```

Comment:
This code describes the types of widgets that can show up on a given screen type. For now, they're almost identical, but that's likely to change in the future, so in this case the duplication is mostly coincidental and (IMO) does not warrant extracting to a common function.

Reply:
This seems like a case where `defoverridable` could be useful — initially all the generators could just `super`, but then could be tweaked or replaced to do different things later.

Reply:
I am resolving this as "won't do". There is no concept of a "base screen type" to abstract to. Once we implement pre-fare screens, the only common element between all types will be the header, and even there we have different types of header (station name, headsign, alert variant on DUPs) on different screens. It might make sense to think of this function as more of a hard-coded configuration for each screen type than a piece of logic that makes sense to be abstracted.